### PR TITLE
UI controls to switch update track were added

### PR DESCRIPTION
**Pull request by olmisa**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Puppet UI was updated. Controls to switch update track were added.

## Related PRs or issues

[AB#75362](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/75362)

### CODE REVIEW

**Review by annakocheshkova — CHANGES_REQUESTED** [review]
Let's name controls as in other platforms. "Before next start"

**Comment by olmisa** [comment]
The option already has the name "Before next start". I have made a text auto size



.

File: Assets/Puppet/PuppetAppCenter.cs [hunk 1]
```diff
@@ -287,6 +303,32 @@ public void SetAppSecret(string appSecret)
 #endif
     }
 
+    public void SetUpdateTrack()
+    {
+        int updateTrack = UpdateTrackDropDown.value + 1;
```
Comment (guperrot) [inline Assets/Puppet/PuppetAppCenter.cs]: `+ 1`?
If that works it looks like a hack, we should have more readable code to switch the value.

File: Assets/Puppet/PuppetAppCenter.cs [hunk 2]
```diff
@@ -287,6 +303,32 @@ public void SetAppSecret(string appSecret)
...
+        PlayerPrefs.SetInt(UpdateTrackKey, updateTrack);
+        PlayerPrefs.Save();
+        UpdateTrackCached = updateTrack;
+#if UNITY_ANDROID && !UNITY_EDITOR
+        AndroidUtility.SetPreferenceInt(UpdateTrackAndroidKey, updateTrack);
+#endif
+        if(WhenUpdateTrackCached == (int)WhenUpdateTrackEnum.Now)
```
Comment (guperrot) [inline Assets/Puppet/PuppetAppCenter.cs]: Reformat file

**Review by guperrot — CHANGES_REQUESTED** [review]
.

File: Assets/Puppet/PuppetAppCenter.cs (hunk 1 above)
Comment (olmisa) [reply Assets/Puppet/PuppetAppCenter.cs]: Fixed

File: Assets/Puppet/PuppetAppCenter.cs (hunk 2 above)
Comment (olmisa) [reply Assets/Puppet/PuppetAppCenter.cs]: Fixed

**Review by annakocheshkova — APPROVED** [review]

**Review by guperrot — CHANGES_REQUESTED** [review]
Before next start does not work, it seems to be done too late.
When current track is private, switching to public in before start and restart: browser is still opened.
When current track is public, switching to private before start: I can see the public call being mode before change applied.

**Comment by guperrot** [comment]
@annakocheshkova I wonder if we need AppCenterBehavior drop down for the before start scenario. I feel like the API would work only for runtime changes after start. If we go that route make sure the behavior's default is to not apply a setting as runtime changes are persistent (something like "updated via API" default value I am not sure).

**Comment by olmisa** [comment]
@guperrot yes, to implement working before start scenario AppCenterBehavior should be updated. `Distribute` started from native code before any Unity code could be executed. So, I have created PR https://github.com/microsoft/appcenter-sdk-unity/pull/397 which contains logic for this scenario.

**Review by guperrot — CHANGES_REQUESTED** [review]
We are going to allow only before start so it will only be in the behavior.

**Pull request closed without merging**